src/dbnode/storage/namespace.go

```diff
@@ -1256,7 +1266,7 @@ func (n *dbNamespace) ColdFlush(flushPersist persist.FlushPreparer) error {
 
 	// If repair is enabled we still need cold flush regardless of whether cold writes is
 	// enabled since repairs are dependent on the cold flushing logic.
-	if !n.nopts.ColdWritesEnabled() && !n.nopts.RepairEnabled() {
+	if n.ReadOnly() || !(n.nopts.ColdWritesEnabled() || n.nopts.RepairEnabled()) {
```

Comment:
nit; may be clearer to read as the following so the logic is more explicit
```
repairEnabled := !n.nopts.ColdWritesEnabled() && !n.nopts.RepairEnabled()
if n.ReadOnly() || repairEnabled {
```

Reply:
Will do it, but in a bit different way.